### ISSUE TITLE
Minio console not deploying in the tenant namespace

### ISSUE THREAD

**Issue by hramdin28** [minio/operator#675]
Hi All,

I am using the example https://github.com/minio/operator/blob/master/examples/tenant-encryption.yaml
Everything works fine apart from the minio console.

## Expected Behavior
Minio console should also be deployed

## Current Behavior
After the kubectl apply, in the tenant namespace, there is no minio console service
NAME                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/minio              ClusterIP   10.105.29.122   <none>        443/TCP    4m27s
service/minio-hl           ClusterIP   None            <none>        9000/TCP   4m27s
service/minio-kes-hl-svc   ClusterIP   None            <none>        7373/TCP   76s

If i check the logs of the minio pod operator in the minio-operator namespace, i see the below:

2021-06-10 15:51:11.527056 I | 2021/06/10 15:51:11 error.go:154: debugging error: secrets "subnet-license" not found
2021-06-10 15:51:11.529880 I | 2021/06/10 15:51:11 admin_tenants.go:408: secrets "minio" not found
2021-06-10 15:51:11.531298 I | 2021/06/10 15:51:11 admin_subscription.go:308: subnet secret doesn't contain jwt license
2021-06-10 15:51:11.533502 I | 2021/06/10 15:51:11 admin_tenants.go:449: services "minio-console" not found
2021-06-10 15:51:31.250168 I | 2021/06/10 15:51:31 admin_tenants.go:408: secrets "minio" not found
2021-06-10 15:51:31.251307 I | 2021/06/10 15:51:31 admin_subscription.go:308: subnet secret doesn't contain jwt license
2021-06-10 15:51:31.253554 I | 2021/06/10 15:51:31 admin_tenants.go:449: services "minio-console" not found

**Comment by nitisht** [comment]
can you delete current tenant (`kubectl delete -f tenant-encryption.yaml`) and start fresh with https://github.com/minio/operator/blob/master/examples/tenant.yaml file?

**Comment by hramdin28** [comment]
Hi,

I did what you asked and again all pods and services work fine but again the minio console pod and service do not show up.

**Comment by nitisht** [comment]
can you paste the operator pod logs?

**Comment by hramdin28** [comment]
0611 10:01:19.190121       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
I0611 10:04:56.966964       1 status.go:54] Hit conflict issue, getting latest version of tenant
I0611 10:04:57.001794       1 main-controller.go:985] Deploying pool pool-0
I0611 10:04:58.075231       1 main-controller.go:990] Deploying pool failed pool-0
E0611 10:04:58.075282       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': MinIO is not ready
I0611 10:05:06.953721       1 main-controller.go:985] Deploying pool pool-0
I0611 10:05:06.960136       1 minio.go:38] Generating private key
I0611 10:05:06.960203       1 minio.go:51] Generating CSR with CN=*.minio-hl.minio-tenant-1.svc.cluster.local
I0611 10:05:06.977125       1 csr.go:145] Start polling for certificate of csr/minio-minio-tenant-1-csr, every 5s, timeout after 20m0s
I0611 10:05:11.983256       1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
E0611 10:05:11.989220       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': waiting for minio cert
I0611 10:06:06.977267       1 main-controller.go:985] Deploying pool pool-0
I0611 10:06:07.045000       1 csr.go:145] Start polling for certificate of csr/minio-console-minio-tenant-1-csr, every 5s, timeout after 20m0s
I0611 10:06:12.049996       1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
E0611 10:06:12.054385       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': waiting for console cert
Get "https://minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local:9000/minio/admin/v3/info": dial tcp: lookup minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local on 10.96.0.10:53: no such host
I0611 10:08:59.367131       1 status.go:54] Hit conflict issue, getting latest version of tenant
I0611 10:08:59.377212       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
Get "https://minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local:9000/minio/admin/v3/info": dial tcp: lookup minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local on 10.96.0.10:53: no such host
I0611 10:12:31.908269       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
Get "https://minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local:9000/minio/admin/v3/info": dial tcp: lookup minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local on 10.96.0.10:53: no such host
I0611 10:14:17.956895       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
Get "https://minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local:9000/minio/admin/v3/info": dial tcp: lookup minio-pool-0-0.minio-hl.minio-tenant-1.svc.cluster.local on 10.96.0.10:53: no such host
I0611 10:15:43.200004       1 status.go:54] Hit conflict issue, getting latest version of tenant
E0611 10:15:43.201178       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': tenants.minio.min.io "minio" not found
E0611 10:15:48.202203       1 main-controller.go:791] Tenant 'minio-tenant-1/minio' in work queue no longer exists
I0611 10:15:48.202242       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
I0611 10:18:02.251197       1 status.go:54] Hit conflict issue, getting latest version of tenant
I0611 10:18:02.290849       1 main-controller.go:985] Deploying pool ss-0
I0611 10:18:03.355215       1 main-controller.go:990] Deploying pool failed ss-0
E0611 10:18:03.355247       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': MinIO is not ready
I0611 10:18:12.235913       1 main-controller.go:985] Deploying pool ss-0
I0611 10:18:12.243615       1 minio.go:38] Generating private key
I0611 10:18:12.243693       1 minio.go:51] Generating CSR with CN=*.minio-hl.minio-tenant-1.svc.cluster.local
I0611 10:18:12.259220       1 csr.go:145] Start polling for certificate of csr/minio-minio-tenant-1-csr, every 5s, timeout after 20m0s
I0611 10:18:17.265796       1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
E0611 10:18:17.269033       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': waiting for minio cert
I0611 10:19:12.265155       1 main-controller.go:985] Deploying pool ss-0
I0611 10:19:12.304490       1 csr.go:145] Start polling for certificate of csr/minio-console-minio-tenant-1-csr, every 5s, timeout after 20m0s
I0611 10:19:17.310892       1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
E0611 10:19:17.314240       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': waiting for console cert
Server not initialized, please try again.
I0611 10:21:22.103674       1 status.go:54] Hit conflict issue, getting latest version of tenant
I0611 10:21:22.111509       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'
Get "https://minio-ss-0-0.minio-hl.minio-tenant-1.svc.cluster.local:9000/minio/admin/v3/info": dial tcp: lookup minio-ss-0-0.minio-hl.minio-tenant-1.svc.cluster.local on 10.96.0.10:53: no such host
I0611 10:24:17.864598       1 status.go:54] Hit conflict issue, getting latest version of tenant
E0611 10:24:17.865625       1 main-controller.go:752] error syncing 'minio-tenant-1/minio': tenants.minio.min.io "minio" not found
E0611 10:24:22.866209       1 main-controller.go:791] Tenant 'minio-tenant-1/minio' in work queue no longer exists
I0611 10:24:22.866230       1 main-controller.go:747] Successfully synced 'minio-tenant-1/minio'

**Comment by balorette** [comment]
Facing similar issue:

I0617 12:41:15.119434       1 csr.go:242] Certificate successfully fetched, creating secret with Private key and Certificate
E0617 12:41:15.130400       1 main-controller.go:742] error syncing 'shared-minio/dev-dnp-minio': waiting for console cert
I0617 12:42:10.195746       1 main-controller.go:989] Deploying pool pool-0
I0617 12:42:10.356828       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:42:15.490572       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:42:35.569366       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:42:38.702171       1 http_handlers.go:180] MINIO_ARGS value is https://dev-dnp-minio-pool-0-{0...3}.dev-dnp-minio-hl.shared-minio.svc.cluster.local/export{0...3}
I0617 12:42:43.051892       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:42:46.416550       1 http_handlers.go:180] MINIO_ARGS value is https://dev-dnp-minio-pool-0-{0...3}.dev-dnp-minio-hl.shared-minio.svc.cluster.local/export{0...3}
I0617 12:42:46.630719       1 http_handlers.go:180] MINIO_ARGS value is https://dev-dnp-minio-pool-0-{0...3}.dev-dnp-minio-hl.shared-minio.svc.cluster.local/export{0...3}
I0617 12:42:51.774914       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:43:08.637210       1 http_handlers.go:180] MINIO_ARGS value is https://dev-dnp-minio-pool-0-{0...3}.dev-dnp-minio-hl.shared-minio.svc.cluster.local/export{0...3}
I0617 12:43:23.300800       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:43:28.361915       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'
I0617 12:46:29.201456       1 main-controller.go:737] Successfully synced 'shared-minio/dev-dnp-minio'

**Comment by Dryr** [comment]
We faced the same problem and found a solution

**TLDR:** the **'accessKey'** for the tenant and the console _need_ to be **different**
```
tenants:
  - name: <tenant-name>
    ...
    secrets:
      accessKey: somename
      secretKey: **************
    ...
    console:
      secrets:
        accessKey: someOTHERname
        secretKey: **************
```

**How we found out:**
Nothing in the logs were indicating that something was preventing the console from being deployed
We raised the verbosity level of the minio operator by modifying the starting command to
```
/minio-operator -stderrthreshold 0 -v 5
```

That's when something interesting showed up:
`I0707 14:29:28.129856 1 console.go:139] User is not allowed to be same as admin access key`

So we modified the accessKey of the tenant and console so they would be different (as mentionned at the start of this comment)
We redeployed the tenant, and the console was properly provisioned

Full error log for people that might need it:
```
I0707 14:27:07.453353 1 csr.go:145] Start polling for certificate of csr/operator-minio-after-csr, every 5s, timeout after 20m0s
I0707 14:27:07.487626 1 shared_informer.go:270] caches populated
I0707 14:27:07.487663 1 main-controller.go:673] Starting workers
I0707 14:27:07.688494 1 main-controller.go:1305] Processing object: minio-operator
I0707 14:27:07.701753 1 main-controller.go:1305] Processing object: minio-operator-after-console
I0707 14:27:12.413570 1 main-controller.go:738] Key from workqueue: minio-tenant-after/minio-tenant-after
I0707 14:27:12.459495 1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
I0707 14:27:12.470768 1 main-controller.go:627] Waiting for the operator certificates to be issued waiting for Operator cert
I0707 14:27:12.488077 1 minio.go:48] Creating a new Certificate Signing Request for MinIO Server Certs, cluster "minio-tenant-after/minio-tenant-after"
I0707 14:27:12.488152 1 minio.go:141] Generating private key
I0707 14:27:12.488291 1 minio.go:154] Generating CSR with CN=*.minio-tenant-after-hl.minio-tenant-after.svc.cluster.local
I0707 14:27:12.504135 1 csr.go:145] Start polling for certificate of csr/minio-tenant-after-minio-tenant-after-csr, every 5s, timeout after 20m0s
I0707 14:27:17.512161 1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
I0707 14:27:17.519741 1 main-controller.go:842] Error when consolidating tenant service: waiting for minio cert
E0707 14:27:17.519814 1 main-controller.go:754] error syncing 'minio-tenant-after/minio-tenant-after': waiting for minio cert
I0707 14:27:22.442004 1 main-controller.go:738] Key from workqueue: minio-tenant-after/minio-tenant-after
I0707 14:27:22.477045 1 main-controller.go:656] Starting api server
I0707 14:27:22.477463 1 minio.go:99] Creating a new Cluster IP Service for cluster "minio-tenant-after/minio-tenant-after"
I0707 14:27:22.500711 1 status.go:54] Hit conflict issue, getting latest version of tenant
I0707 14:27:22.519952 1 main-controller.go:853] Creating a new Headless Service for cluster "minio-tenant-after/minio-tenant-after"
I0707 14:27:22.566103 1 main-controller.go:977] Deploying pool ss-0
I0707 14:27:23.643532 1 main-controller.go:982] Deploying pool failed ss-0
E0707 14:27:23.643615 1 main-controller.go:754] error syncing 'minio-tenant-after/minio-tenant-after': MinIO is not ready
I0707 14:27:37.413380 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:27:37.418975 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:27:37.440543 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:07.414555 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:07.420028 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:07.442089 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:16.472725 1 main-controller.go:1305] Processing object: istio-ingressgateway
I0707 14:28:16.503341 1 main-controller.go:1305] Processing object: istio-ingressgateway
I0707 14:28:16.589126 1 main-controller.go:1305] Processing object: istio-ingressgateway
I0707 14:28:22.480272 1 main-controller.go:738] Key from workqueue: minio-tenant-after/minio-tenant-after
I0707 14:28:22.517495 1 main-controller.go:977] Deploying pool ss-0
I0707 14:28:22.555259 1 main-controller.go:1305] Processing object: minio-tenant-after-ss-0
I0707 14:28:22.602194 1 console.go:298] Creating a new Certificate Signing Request for Console Server Certs, cluster "minio-tenant-after/minio-tenant-after"
I0707 14:28:22.626188 1 csr.go:145] Start polling for certificate of csr/minio-tenant-after-console-minio-tenant-after-csr, every 5s, timeout after 20m0s
I0707 14:28:22.655285 1 main-controller.go:1305] Processing object: minio-tenant-after-ss-0
I0707 14:28:27.632239 1 csr.go:170] Certificate successfully fetched, creating secret with Private key and Certificate
I0707 14:28:27.639400 1 main-controller.go:1195] Error checking console state waiting for console cert
E0707 14:28:27.639469 1 main-controller.go:754] error syncing 'minio-tenant-after/minio-tenant-after': waiting for console cert
I0707 14:28:37.415113 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:37.420674 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:37.443462 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:28:40.769441 1 main-controller.go:1305] Processing object: minio-tenant-after-ss-0
I0707 14:28:41.133684 1 main-controller.go:1305] Processing object: minio-tenant-after-ss-0
I0707 14:29:07.417555 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:29:07.421961 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:29:07.444569 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:29:22.532546 1 main-controller.go:738] Key from workqueue: minio-tenant-after/minio-tenant-after
I0707 14:29:22.895473 1 console.go:139] User is not allowed to be same as admin access key
I0707 14:29:22.908097 1 status.go:54] Hit conflict issue, getting latest version of tenant
I0707 14:29:22.930799 1 main-controller.go:749] Successfully synced 'minio-tenant-after/minio-tenant-after'
I0707 14:29:27.931452 1 main-controller.go:738] Key from workqueue: minio-tenant-after/minio-tenant-after
I0707 14:29:28.129856 1 console.go:139] User is not allowed to be same as admin access key
I0707 14:29:28.129904 1 main-controller.go:749] Successfully synced 'minio-tenant-after/minio-tenant-after'
I0707 14:29:37.418764 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
I0707 14:29:37.423116 1 reflector.go:381] k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: forcing resync
```

**Comment by hramdin28** [comment]
Yes you are right. Using different secretkey and accesskey solved the issue for me. Thanks a lot

**Comment by hramdin28** [comment]
> @hramdin28 Can you tell me where and how to change this access key, can't seem to find anything during tenant installation via GUI for that? or did you apply custom yaml for it?

Hello i used a custom yaml based on the examples. 
Here is what i changed.
---
apiVersion: v1
kind: Secret
metadata:
  name: minio-creds-secret
type: Opaque
data:
  accesskey: aaaa
  secretkey: bbbb
---
apiVersion: v1
kind: Secret
metadata:
  name: console-secret
type: Opaque
data:
  CONSOLE_PBKDF_PASSPHRASE: xxxx
  CONSOLE_PBKDF_SALT: xxxx
  CONSOLE_ACCESS_KEY: yyyy
  CONSOLE_SECRET_KEY: zzzz 

Notice the access keys and secret keys are different. If i use same values then the minio-console do  not start
